docs/build/build-transaction-construction.md

```diff
@@ -35,6 +35,27 @@ function from the Balances pallet will take:
 - `dest`: Destination address
 - `#[compact] value`: Number of tokens (compact encoding)
 
+**Serialized transaction format**
+
+Before being submitted, transactions are serialized. Serialized transactions are hex encoded SCALE-encoded bytes with the following format:
+- Compact encoded number of SCALE-encoded bytes following this.
+- 1 bit: it is a 0 if no signature is present, or a 1 if it is.
+- 7 bits: the transaction protocol version.
```

Comment:
This is different from the `transaction_version`. It's confusing, but at least note it here.

Reply:
I though these 7 bits was exactly `transaction_version` as described [here](https://docs.substrate.io/v3/runtime/upgrades/).

What are they then? I will update this accordingly.

Reply:
Nope, so this is a bit lower level, it's like the version of "what an extrinsic is" _to Substrate_, while `transaction_version` refers to backwards incompatible changes _to a specific runtime,_ namely when you break pallet/call indexing, function signatures (including types), or types in the serialized transaction itself (e.g. `AccountId` or `SignedExtension`s).

`transaction_version` came much later than `extrinsic_version`, and was a bit of a naming collision that most people just didn't notice. Unless there is a major major major change to Substrate, `extrinsic_version` will basically always `= 4`, and a change would mean basically needing to update everything.

You can see a few examples of its use in txwrapper:

- [Decoding extrinsics](https://github.com/paritytech/txwrapper-core/blob/d590b0912ee0bf9f357233024c064138cff36cd2/packages/txwrapper-core/src/core/decode/decodeSigningPayload.ts#L38) (notice `transaction_version` at L54)
- [The definition of a "method"](https://github.com/paritytech/txwrapper-core/blob/3cc7bf3e63f1e88d231454a98307afe4ab798b27/packages/txwrapper-core/src/core/method/defineMethod.ts#L106)
- [Explained a bit in an example](https://github.com/paritytech/txwrapper-core/blob/b3d7fc1ea49244cbda18d08f14a5d2cdd0b2e78c/packages/txwrapper-core/src/core/construct/createSigningPayload.ts#L27)

Reply:
I updated the description to "extrinsic version" also based on this PR: https://github.com/paritytech/substrate/commit/6bcc3851909284f934461e4dc45fec72ce1cb525